### PR TITLE
Added useSuspenseQuery hook, remove top-level suspense: true

### PR DESCRIPTION
**Pull request by krishnaglick**
## What
Removed the top-level Suspense configuration in favor of a useSuspenseQuery hook, swapping-in-place where appropriate.
No need for `as QueryObserverSuccessResult<T>).data` any more.

### CODE REVIEW

File: airbyte-webapp/src/hooks/services/useDocumentation.ts [hunk 1]
```diff
@@ -27,6 +27,5 @@ export const useDocumentation = (documentationUrl: string): UseDocumentationResu
     refetchOnMount: false,
     refetchOnWindowFocus: false,
     retry: false,
-    suspense: false,
```
Comment (krishnaglick) [inline airbyte-webapp/src/hooks/services/useDocumentation.ts:30]: Suspense defaults to false now

File: airbyte-webapp/src/services/connector/DestinationDefinitionSpecificationService.tsx [hunk 2]
```diff
@@ -42,7 +39,6 @@ export const useGetDestinationDefinitionSpecificationAsync = (
 
   const escapedId = id ?? "";
   return useQuery(destinationDefinitionSpecificationKeys.detail(escapedId), () => service.get(escapedId), {
-    suspense: false,
```
Comment (krishnaglick) [inline airbyte-webapp/src/services/connector/DestinationDefinitionSpecificationService.tsx:45]: Suspense defaults to false now

File: airbyte-webapp/src/services/connector/SourceDefinitionSpecificationService.tsx [hunk 3]
```diff
@@ -42,7 +39,6 @@ export const useGetSourceDefinitionSpecificationAsync = (
 
   const escapedId = id ?? "";
   return useQuery(sourceDefinitionSpecificationKeys.detail(escapedId), () => service.get(escapedId), {
-    suspense: false,
```
Comment (krishnaglick) [inline airbyte-webapp/src/services/connector/SourceDefinitionSpecificationService.tsx:45]: Suspense defaults to false now

File: airbyte-webapp/src/services/connector/useSuspenseQuery.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+>(
+  queryKey: TQueryKey,
+  queryFn: QueryFunction<TQueryFnData, TQueryKey>,
+  options: Readonly<
+    Omit<UseQueryOptions<TQueryFnData, TError, TData, TQueryKey>, "queryKey" | "queryFn" | "suspense">
+  > = {}
+) {
+  return useQuery<TQueryFnData, TError, TData, TQueryKey>(queryKey, queryFn, {
+    ...options,
+    suspense: true,
```
Comment (krishnaglick) [inline airbyte-webapp/src/services/connector/useSuspenseQuery.ts:19]: Force suspense for the hook

File: airbyte-webapp/src/services/connector/useSuspenseQuery.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  }).data as typeof options extends Disabled ? TData | undefined : TData;
```
Comment (krishnaglick) [inline airbyte-webapp/src/services/connector/useSuspenseQuery.ts:20]: We sometimes disable the call due to the rule of hooks. This lets typescript correctly infer the return type.

File: airbyte-webapp/src/views/common/StoreProvider.tsx [hunk 6]
```diff
@@ -5,7 +5,6 @@ import { ReactQueryDevtools } from "react-query/devtools";
 const queryClient = new QueryClient({
   defaultOptions: {
     queries: {
-      suspense: true,
```
Comment (krishnaglick) [inline airbyte-webapp/src/views/common/StoreProvider.tsx:8]: Defaults to false now since we have the new suspense hook.

**Review by timroes — APPROVED** [review]
Tested several parts of the UI (Cloud mode), nothing seemed to break. Code LGTM

**Pull request merged**